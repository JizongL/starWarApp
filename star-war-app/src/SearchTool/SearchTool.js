import React from 'react'
import starWarContext from '../context/starWarContext'

class SearchTool extends React.Component{
  state={
    searchTerm:'',
    searchCategory:'',
    loading:false,
    error:null
  }
  static contextType = starWarContext
  
  doFetch=(searchTerm,category)=>{
    this.setState({loading:true})
    fetch(`https://swapi.co/api/${category}/?search=${searchTerm}`).then(
      res=> 
      {
        if(!res.ok){
          throw new Error(res.json())
        }
        return res.json()
      }
    ).then(data=> {this.setState({
      data:data,
      loading:false
    },this.context.dataHandle(data));
  })

    .catch(err=>this.setState({error:err}))
  }

  handleInput=(value)=>{
    this.setState({searchTerm:value})
    
  }

  handleSelect=(value)=>{
    
    this.setState({searchCategory:value})
    
  }

  handleSubmit=(e)=>{
    e.preventDefault()
    let searchTerm = this.state.searchTerm;
    let category = this.state.searchCategory
    //console.log(searchTerm,category,'test search and cat')
    this.doFetch(searchTerm,category)
    
  }

  render(){
    
    const options=this.context.search.map((option,index)=>
      <option key={index}>{option}</option>)
     // console.log(this.state,'test data')
    return(
      <div>
      <form className='search-form'
        onSubmit ={(e)=>this.handleSubmit(e)}
      >
      <input id='search-input' placeholder='..Search'
      onChange={(e)=>this.handleInput(e.target.value)}/>
      <select onChange={(e)=>this.handleSelect(e.target.value)}>
        <option >choose category</option>
        {options}
        </select>
      <button>Search</button>
      </form>
      
      {this.state.loading && <span>loading...</span>}
      {this.state.error && <span>{this.state.error}</span>}
      </div>
    )
  }
}

export default SearchTool