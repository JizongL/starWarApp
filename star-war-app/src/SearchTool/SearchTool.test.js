import React from 'react'
import ReactDOM from 'react-dom'
import SearchTool from './SearchTool'
import renderer from 'react-test-renderer'

describe('test Search Tool',()=>{
  it('renders without crashing',()=>{
    const div= document.createElement('div')
    ReactDOM.render(<SearchTool/>,div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  it('redners the UI as expected',()=>{
    const tree = renderer
    .create(<SearchTool/>).toJSON;
    expect(tree).toMatchSnapshot();
  
  });
  
  
})
