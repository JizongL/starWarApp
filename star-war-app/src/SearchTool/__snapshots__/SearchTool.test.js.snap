// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redners the UI as expected 1`] = `[Function]`;

exports[`test Search Tool redners the UI as expected 1`] = `[Function]`;
